packages/core/src/props/index.ts

```diff
@@ -75,3 +75,15 @@ export type GlobalHeadProps = {
 	globalStylesheet?: string | null;
 	renderedFontFaces?: string | null;
 };
+
+type DOMNode = {
+	type: string;
+	name?: string;
+	children?: DOMNode[];
+	attribs?: Record< string, string >;
+	data?: string;
+};
+
+export type HTMLReactParserOptions = {
+	replace?: ( domNode: DOMNode ) => React.ReactElement | string | null | void;
+};
```

Comment:
I don't like the idea of manually maintaining this.

Not entirely sure how this works but can html-react-parser be a devDep so we can just recycle their type?